strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BrowseController.java

```diff
@@ -174,4 +171,277 @@
+        {
+            try
+            {
+                DirectoryContent content = new DirectoryContent(baseDir);
+                URL url = artifactResolutionService.resolveResource(storageId, repositoryId, subPath);
+                generateHTML(request, response, content, url);
+            }
+            catch (Exception e)
+            {
+                logger.debug("Unable to generate directory listing for " +
+                        "/" + storageId + "/" + repositoryId + "/" + path, e);
+
+                response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
+            }
+
+            return;
+        }
+        return;
+    }    
+    
+    protected void generateHTML(HttpServletRequest request,
+                                HttpServletResponse response,
+                                DirectoryContent content,
+                                URL url)
+    {
+        if(content == null)
+        {
+            response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
+            logger.debug("Could not retrive /storages base directory");
+            return;
+        }
+                
+        String requestUri = request.getRequestURI();
+        
+        if (!requestUri.endsWith("/"))
+        {
+            try
+            {
+                response.sendRedirect(requestUri + "/");
+            }
+            catch (IOException e)
+            {
+                logger.debug("Error redirecting to " + requestUri + "/");
+            }
+            return;
+        }
+
+        try
+        {
+            StringBuilder sb = new StringBuilder();
+            sb.append("<html>");
+            sb.append("<head>");
+            sb.append(
+                    "<style>body{font-family: \"Trebuchet MS\", verdana, lucida, arial, helvetica, sans-serif;} table tr {text-align: left;}</style>");
+            sb.append("<title>Index of " + request.getRequestURI() + "</title>");
+            sb.append("</head>");
+            sb.append("<body>");
+            sb.append("<h1>Index of " + request.getRequestURI() + "</h1>");
+            sb.append("<table cellspacing=\"10\">");
+            sb.append("<tr>");
+            sb.append("<th>Name</th>");
+            sb.append("<th>Last modified</th>");
+            sb.append("<th>Size</th>");
+            sb.append("<th>Description</th>");
+            sb.append("</tr>");
+            sb.append("<tr>");
+            sb.append("<td colspan=4><a href=\"..\">..</a></td>");
+            sb.append("</tr>");
+           
+            for (FileContent fileContent : content.getDirectories())
+            {
+                appendFile(sb, fileContent, requestUri, true, url);
+            }
+                        
+            if(url == null && !content.getFiles().isEmpty())
+            {
+                logger.debug("Files are present outside a repository");
+                response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
```

Comment:
can you please follow [How-To-Implement-Spring-Controllers](https://github.com/strongbox/strongbox/wiki/How-To-Implement-Spring-Controllers) guide

Reply:
+1